cmd/kubeapps-apis/server/plugins.go

```diff
@@ -81,26 +98,27 @@ func (s *pluginsServer) GetConfiguredPlugins(ctx context.Context, in *plugins.Ge
 }
 
 // registerPlugins opens each plugin, looks up the register function and calls it with the registrar.
-func registerPlugins(pluginPaths []string, grpcReg grpc.ServiceRegistrar, gwArgs gwHandlerArgs) ([]*plugins.Plugin, error) {
+func (s *pluginsServer) registerPlugins(pluginPaths []string, grpcReg grpc.ServiceRegistrar, gwArgs gwHandlerArgs) ([]*plugins.Plugin, error) {
 	pluginDetails := []*plugins.Plugin{}
 	for _, pluginPath := range pluginPaths {
 		p, err := plugin.Open(pluginPath)
 		if err != nil {
 			return nil, fmt.Errorf("unable to open plugin %q: %w", pluginPath, err)
 		}
 
-		if err = registerGRPC(p, pluginPath, grpcReg); err != nil {
+		var pluginDetail *plugins.Plugin
+		if pluginDetail, err = getPluginDetail(p, pluginPath); err != nil {
 			return nil, err
+		} else {
+			pluginDetails = append(pluginDetails, pluginDetail)
 		}
 
-		if err = registerHTTP(p, pluginPath, gwArgs); err != nil {
+		if err = s.registerGRPC(p, pluginDetail, grpcReg); err != nil {
 			return nil, err
 		}
 
-		if pluginDetail, err := getPluginDetail(p, pluginPath); err != nil {
+		if err = registerHTTP(p, pluginDetail, gwArgs); err != nil {
```

Comment:
`registerGRPC` was only updated to a method of the `pluginServer` because I needed to store state while registering each plugin. Yep, I could update `registerHTTP` to be a method as well for consistency, but my preference is for stateless functions where possible (easier to reason about and obvious that there's no state of pluginServer being used).